sdb/__init__.py

```diff
@@ -191,3 +175,20 @@ def invoke(prog: drgn.Program, first_input: Iterable[drgn.Object],
         if shell_cmd is not None:
             sys.stdout = old_stdout
             shell_proc.wait()
+
+
+def get_first_type(objs: Iterable[drgn.Object]
+                  ) -> Tuple[drgn.Type, Iterable[drgn.Object]]:
+    """
+    Determine the type of the first object in the iterable. The first element
+    in the iterable will be consumed. Therefore, a tuple is returned with the
+    type and a new iterable with the same values as the specified iterable.
+    Callers must use the returned iterable in place of the specified one.
+
+    e.g.: first_type, objs = sdb.get_first_type(objs)
+    """
+    iterator = iter(objs)
+    first_obj = next(iterator, None)
+    if first_obj is None:
+        return None, []
+    return first_obj.type_, itertools.chain([first_obj], iterator)
```

Comment:
Looking more into this it seems like it kinda works as we want it to as long as we only peek at one element -> https://docs.python.org/3.6/library/itertools.html#itertools.tee
I'm not sure how I feel about the complexity/cleverness of this but this function is pretty small so I can see the point both ways.

Reply:
I think that we would want to keep the interface to `get_first_type`, so we'd need to return a tuple, because I think it's better to encapsulate the `next(iter())` or `tee`.  e.g. caller continue to do:
```
first_type, objs = sdb.get_first_type(objs)
```
rather than
```
objs, itercopy = itertools.tee(objs)
first_type = sdb.get_first_type(itercopy)
```

Given that, I could go either way with the internals of `get_first_type`.

Reply:
`itertools.chain` is more intuitive - let's leave the code as it is.